### ISSUE TITLE
blockchain v2: node might panic with nil last commit

### ISSUE THREAD

**Issue by cmwaters** [tendermint/tendermint#5396]
## Summary

We need more graceful error handling

## Problem Definition

If on fast sync, a peer sends the v2 reactor a block with a nil last commit, we do not afaik perform a `ValidateBasic()` thus the block can be used in `rProcessBlock` in `processor.go` which runs:

https://github.com/tendermint/tendermint/blob/ebfe0cbb8cc335eee8bf5ee2ab9bdce66ffb619f/blockchain/v2/processor.go#L154

This in turn runs `VerifyCommitLight()` which will cause a nil pointer dereference panic here

```
func (vals *ValidatorSet) VerifyCommitLight(chainID string, blockID BlockID,
	height int64, commit *Commit) error {

	if vals.Size() != len(commit.Signatures) {
		return NewErrInvalidCommitSignatures(vals.Size(), len(commit.Signatures))
	}

        . . .
}
```

## Proposal

`VerifyCommitLight` should have a nil check for the commit at the start or we ensure that it's not nil before running it

Generally we should try avoid panics in the blockchain reactor and handle them gracefully.

https://github.com/tendermint/tendermint/blob/ebfe0cbb8cc335eee8bf5ee2ab9bdce66ffb619f/blockchain/v2/processor.go#L163-L167

I would propose having a validate basic in place when a block is received which if it fails then drops the peer. Alternatively, as `ApplyBlock` also runs a validate basic we could handle it there by also dropping the peer on error and then trying again (but I prefer the first option).

Note: I think this case might also apply with the other versions

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

**Comment by melekes** [comment]
> I would propose having a validate basic in place when a block is received which if it fails then drops the peer. Alternatively, as ApplyBlock also runs a validate basic we could handle it there by also dropping the peer on error and then trying again (but I prefer the first option).

👍 for the first option

**Comment by cmwaters** [comment]
Closing this as we do run a validate basic when we convert from proto. Nil checks where also implemented in #5418
